src/nfa/nfa-incl.cc

```diff
@@ -212,3 +212,30 @@ bool Mata::Nfa::is_incl(
 
 	return algo(smaller, bigger, alphabet, cex, params);
 } // is_incl }}}
+
+bool Mata::Nfa::equivalence_check(const Nfa& lhs, const Nfa& rhs, const Alphabet& alphabet, const StringDict& params)
+{
+    if (is_incl(lhs, rhs, alphabet, params))
+    {
+        if (is_incl(rhs, lhs, alphabet, params))
+        {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool Mata::Nfa::equivalence_check(const Nfa& lhs, const Nfa& rhs, const StringDict& params)
```

Comment:
I would probably add note here, that this is not exactly efficient equivalence check, and the above function which takes Alphabet should be prefered, i.e. have the alphabet computed only once (currently, the ad-hoc construction of alphabet is O(trans)*O(set.insert), which could potentially hurt). For debug, I guess it's OK though, so I would let it here, but warn potential user.

Reply:
The current state of handling alphabets in Mata is only temporal. I expect more changes to come for these operations directly working with alphabet.

For now, though, the suggestion is reasonable, and I will add a note to warn the user. Thank you for the suggestion.